src/picongpu/include/particles/ionization/byCollision/ThomasFermi/ThomasFermi_Impl.hpp

```diff
@@ -272,20 +276,39 @@ namespace ionization
                 /* energy density in sim units */
                 float_X kinEnergyDensity = kinEnergyV[0];
 
-                /* this is the point where actual ionization takes place */
+                /* Thomas-Fermi calculation step:
+                 * Determines the new average charge state for each ion under
+                 * LTE conditions. Returns the new number of bound electrons
+                 * according to this average charge state and a Monte-Carlo
+                 * step which decides for an integer number of macro electrons.
+                 */
                 IonizationAlgorithm ionizeAlgo;
                 float_X newBoundElectrons = ionizeAlgo(
                     kinEnergyDensity, density,
                     particle, this->randomGen()
                     );
 
-                particle[boundElectrons_] = newBoundElectrons;
-
-                /* safety check to avoid double counting since recombination is not yet implemented */
+                /* Safety Check: ionize only if the number of bound electrons
```

Comment:
Safety check is generic and misleading.

You are doing a detailed balance and now voluntarily ignore re-combinations.
```C++
/* Only account for ionization: we only increase the charge state of an ion
  * if necessary, but ignore recombination of electrons as prediced by the implemented
  * detailed balance algorithm.
  */
```

This is also the reason why I would change the result of the TF implementation *inside* the functor with `if (prevBoundElectrons > newBoundElectrons)` instead of throwing this back all the way to here and making a dozen of comments that the implementation is not what one expects.